pkg/kubectl/cmd/util/printing.go

```diff
@@ -105,36 +105,19 @@ func ValidateOutputArgs(cmd *cobra.Command) error {
 	return nil
 }
 
-// PrinterForCommand returns the default printer for this command.
-// Requires that printer flags have been added to cmd (see AddPrinterFlags).
-func PrinterForCommand(cmd *cobra.Command, mapper meta.RESTMapper, typer runtime.ObjectTyper, decoders []runtime.Decoder) (printers.ResourcePrinter, bool, error) {
-	outputFormat := GetFlagString(cmd, "output")
-
-	// templates are logically optional for specifying a format.
-	// TODO once https://github.com/kubernetes/kubernetes/issues/12668 is fixed, this should fall back to GetFlagString
-	templateFile, _ := cmd.Flags().GetString("template")
-	if len(outputFormat) == 0 && len(templateFile) != 0 {
-		outputFormat = "template"
+// PrinterForCommand returns the printer for the outputOptions (if given) or
+// returns the default printer for the command. Requires that printer flags have
+// been added to cmd (see AddPrinterFlags).
+// TODO: remove the dependency on cmd object
+func PrinterForCommand(cmd *cobra.Command, outputOpts *printers.OutputOptions, mapper meta.RESTMapper, typer runtime.ObjectTyper, decoders []runtime.Decoder) (printers.ResourcePrinter, bool, error) {
+
+	if outputOpts == nil {
+		outputOpts = extractOutputOptions(cmd)
 	}
 
-	templateFormat := []string{
-		"go-template=", "go-template-file=", "jsonpath=", "jsonpath-file=", "custom-columns=", "custom-columns-file=",
-	}
-	for _, format := range templateFormat {
-		if strings.HasPrefix(outputFormat, format) {
-			templateFile = outputFormat[len(format):]
-			outputFormat = format[:len(format)-1]
-		}
-	}
-
-	// this function may be invoked by a command that did not call AddPrinterFlags first, so we need
-	// to be safe about how we access the allow-missing-template-keys flag
-	allowMissingTemplateKeys := false
-	if cmd.Flags().Lookup("allow-missing-template-keys") != nil {
-		allowMissingTemplateKeys = GetFlagBool(cmd, "allow-missing-template-keys")
-	}
 	printer, generic, err := printers.GetStandardPrinter(
```

Comment:
I think we can simplify the signature of `GetStandardPrinter()` by passing in `outputOptions` instead of `outputFormat, templateFile,allowMissingTemplateKeys`.